Assets/SEE/UI/Tooltip.cs

```diff
@@ -0,0 +1,359 @@
+﻿using System;
+using System.Linq;
+using DG.Tweening;
+using Michsky.UI.ModernUIPack;
+using SEE.GO;
+using SEE.Utils;
+using TMPro;
+using UnityEngine;
+
+namespace SEE.UI
+{
+    /// <summary>
+    /// A box containing additional information which usually appears after hovering over any element for an
+    /// amount of time and disappears when moving the pointer away from the element.
+    ///
+    /// It should be used via its static methods <see cref="ActivateWith"/> and <see cref="Deactivate"/>.
+    /// You should not add this component to any game object yourself.
+    ///
+    /// Call <see cref="ActivateWith"/> when the mouse cursor enters the region of an object which should
+    /// display a tooltip, and call <see cref="Deactivate"/> when the mouse cursor leaves the region.
+    /// This class will render a tooltip on the screen with the given text if the mouse cursor stays in the region
+    /// without moving for a certain amount of time.
+    /// </summary>
+    public class Tooltip : PlatformDependentComponent
+    {
+        // TODO: Make this work on VR.
+
+        /// <summary>
+        /// The kinds of behavior the tooltip can have after it has been shown.
+        /// </summary>
+        public enum AfterShownBehavior
+        {
+            /// <summary>
+            /// The tooltip will reappear after it has been shown (assuming no mouse movement for a while).
+            /// </summary>
+            Reappear,
+
+            /// <summary>
+            /// The tooltip will hide until the text changes.
+            /// </summary>
+            HideUntilChanged,
+
+            /// <summary>
+            /// The tooltip will hide until it is activated (i.e., <see cref="ActivateWith"/> is called) again.
+            /// </summary>
+            HideUntilActivated
+        }
+
+        /// <summary>
+        /// The path to the prefab for the tooltip game object.
+        /// Will be added as a child to the <see cref="Canvas"/>.
+        /// </summary>
+        private const string tooltipPrefab = "Prefabs/UI/Tooltip";
+
+        /// <summary>
+        /// Name of the tooltip game object.
+        /// </summary>
+        private const string tooltipName = "SEETooltip";
+
+        /// <summary>
+        /// The time it shall take to fade in the tooltip.
+        /// </summary>
+        private const float fadeInDuration = 0.3f;
+
+        /// <summary>
+        /// The time it will take to fade out.
+        /// </summary>
+        private const float fadeOutDuration = fadeInDuration / 2;
+
+        /// <summary>
+        /// The time to wait before showing the tooltip.
```

Comment:
What is the unit for time?

Reply:
I fixed it assuming that the unity is seconds (looks like the DoTween API uses that unit).